docs: catalog maintenance improvements (#1379)
bareos/bareos: docs/manuals/source/TasksAndConcepts/CatalogMaintenance.rst

::

   Size = 154 * No. Systems * (100,000 * 4 + 10,000 * 26)

Where we have assumed four weeks in a month and 26 incremental backups per month. This would give the following:

::

   Size = 154 * 2 * (100,000 * 4 + 10,000 * 26) = 203,280,000 bytes
   Indexes Size = (154 * 2 * (100,000 * 4 + 10,000 * 26))/3 = 67,760,000

So for the above two systems, we should expect to have a database size of approximately 270 Megabytes including the indexes.
Of course, this will vary according to how many files are actually backed up.

You will note that the File table (containing the file attributes) make up the large bulk of the number of records as well as the space used.

Without proper setup and maintenance, your Catalog may continue to grow indefinitely read carefully the following sections for planning free space and autovacuuming.

.. _FreeSpacePostgres:

Free space needed with PostgreSQL Database
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To ensure that all PostgreSQL maintenance operations like vacuuming and reindexing roll out smoothly we highly recommend not to fill the disk containing the postgresql data directory to more than 50% during normal operation.

> If this is not possible, you might need to make more space available at least temporary when the database is being upgraded to a new schema version.```

Reviewer: ```suggestion
If this is not possible, you might need to make more space available at least temporary when the database is being upgraded to a new schema version.
```

Normal Bareos operation can create huge temp file requiring free space.

> Upgrading your PostgreSQL version to a new major, will from time to time impose a reindex operation which will use temp space too, especially if option :command:`concurrently` is used.

Reviewer: ```suggestion
Upgrading your PostgreSQL version to a new Bareos major version will, from time to time, impose a reindex operation which will use temp space too, especially if option :command:`concurrently` is used.
```

Author: In that case, I was not thinking about major Bareos version (we mostly never do reindex) but obviously new Major PostgreSQL. So I redo the wording to make that more evident.

```suggestion
Upgrading to a new PostgreSQL major version, will from time to time, impose a reindex operation which will use temp space too, especially if option :command:`concurrently` is used.
```

   single: Database; PostgreSQL
   single: PostgreSQL

.. _FreeSpacePostgres:

Free space needed with PostgreSQL Database
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To ensure that all PostgreSQL maintenance operations like vacuuming and reindexing roll out smoothly we highly recommend to keep at least the size of your database available as free space.

Reviewer: ```suggestion
To ensure that all PostgreSQL maintenance operations like vacuuming and reindexing roll out smoothly we highly recommend not to fill the disk containing the postgresql data directory to more than 50% during normal operation. 
If this is not possible, you might need to make more space available at least temporary when the database is being upgraded to a new schema version.```

Author: Doesn't that change a bit the perception? Like then when I'm doing a vacuum full (normal operation) the space will be filled up, what is important is that before the operation start the space available is at least the size of the db. (in fact the size of the biggest table and indexes).

> Bareos has some routines creating huge temp file for its operation, which also needs free space.

Reviewer: Which routines?

Author: bvfs_update, big accurate file list construction, virtual full and so on.

Author: Normally resolved with new wording.

> Bareos has some routines creating huge temp file for its operation, which also needs free space.

Author: ```suggestion
Normal Bareos operation can create huge temp file requiring free space.
```

   }

It is preferable to write/send the :ref:`bootstrap <BootstrapChapter>` file to another computer. It will allow you to quickly recover the database backup should that be necessary. If you do not have a bootstrap file, it is still possible to recover your database backup, but it will be more work and take longer.

> 

Reviewer: Is using the PGDG Packages really what we  recommend?
Why are people paying for the OS like RHEL where they get more recent postgres versions from their OS vendor?

Author: Because you don't want to run 9.2 or 9.6 on EL7 for example.
Because you have a beefy server and you will be better serve by a 15 with // process than the default 12 or 13 given by RHEL.

At least now rhel as start to move a bit forward, so people will have less and less reason to move to pgdg, but still there will be some case. If they use them, then we should just explain how to succeed on first run with bareos.

Author: Ok re-wording done.

.. _PGDG:

Provide postgresql.service with PGDG packages
---------------------------------------------

.. index::
   single: PGDG; Bareos database
   single: systemd; bareos-dir.service
   single: systemd; bareos-director.service
   single: systemd; postgresql-XX.service

> On several distributions, it maybe advisable to use more recent version of PostgreSQL.
> In such case we recommend to rely on the official :strong:`PostgreSQL Global Development Group`
> aka :strong:`PGDG` packages.
> They can be obtained from repository referenced on page `download PostgreSQL <https://www.postgresql.org/download/>`_\.
> 
> While the packages will work fine with Bareos, their politic to never create conflicts with distributor packages,
> their installation will not offer the warranty that postgresql is started before the |dir| daemon.

Author: ```suggestion
If you are using packages from :strong:`PostgreSQL Global Development Group` aka :strong:`PGDG` the delivered systemd service needs to be edited to add the standard `postgresql.service` alias which is required to start |dir| after postgresql.
```
